executor/src/main/java/io/kestra/executor/ExecutorService.java

```diff
@@ -1171,7 +1171,7 @@ private Execution addDynamicTaskRun(Execution execution, Supplier<Flow> flow, Wo
             }
         }
 
-        return taskRuns.size() > execution.getTaskRunList().size() ? execution.withTaskRunList(taskRuns) : null;
+        return taskRuns.size() > ListUtils.emptyOnNull(execution.getTaskRunList()).size() ? execution.withTaskRunList(taskRuns) : null;
```

Comment:
```suggestion
var taskRunList = ListUtils.emptyOnNull(execution.getTaskRunList());
        return taskRuns.size() > taskRunList.size() ? taskRunList : null;
```

If taskRuns is empty, and `execution.getTaskRunList()` is null, it will enter the first case and throw NPE
In case taskRuns can be empty